src/modules/deploy/provider/AWSProvider.ts

```diff
@@ -0,0 +1,103 @@
+import { BaseProvider } from './BaseProvider'
+import AWS from 'aws-sdk'
+
+class AWSProvider extends BaseProvider {
+  public constructor(account: string) {
+    AWS.config.update({ region: 'us-east-2' })
+    super(account)
+  }
+
+  public async listFunctions() {
+    const lambda = new AWS.Lambda()
+    const functions = await lambda.listFunctions().promise()
+    const functionsObj: Record<string, { url: string }> = {}
+
+    functions?.Functions?.forEach((item) => (functionsObj[item.FunctionName!] = { ...item, url: item.Handler! }))
```

Comment:
Maybe investigate the return type and avoid using non-null assertions here

Reply:
This is the [optional chaining operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining), if the object or property are null or undefined, it will just return it.